forge-game/src/main/java/forge/game/staticability/StaticAbilityAttackRestrict.java

```diff
@@ -0,0 +1,53 @@
+package forge.game.staticability;
+
+import forge.game.Game;
+import forge.game.GameEntity;
+import forge.game.ability.AbilityUtils;
+import forge.game.card.Card;
+import forge.game.zone.ZoneType;
+
+public class StaticAbilityAttackRestrict {
+
+    static String MODE = "AttackRestrict";
+
+    static public int globalAttackRestrict(Game game) {
+        int max = Integer.MAX_VALUE;
+        for (final Card ca : game.getCardsIn(ZoneType.STATIC_ABILITIES_SOURCE_ZONES)) {
+            for (final StaticAbility stAb : ca.getStaticAbilities()) {
+                if (!stAb.getParam("Mode").equals(MODE) || stAb.isSuppressed() || !stAb.checkConditions()
+                        || stAb.hasParam("ValidDefender")) {
+                    continue;
+                }
+                int stMax = AbilityUtils.calculateAmount(stAb.getHostCard(),
+                        stAb.getParamOrDefault("MaxAttackers", "1"), stAb);
+                if (stMax < max) {
+                    max = stMax;
+                }
+            }
+        }
+        return max < Integer.MAX_VALUE ? max : -1;
+    }
+
+    static public int attackRestrictNum(GameEntity defender) {
+        final Game game = defender.getGame();
+        for (final Card ca : game.getCardsIn(ZoneType.STATIC_ABILITIES_SOURCE_ZONES)) {
+            for (final StaticAbility stAb : ca.getStaticAbilities()) {
+                if (!stAb.getParam("Mode").equals(MODE) || stAb.isSuppressed() || !stAb.checkConditions()) {
```

Comment:
You need to check also check if it doesn't have ValidDefender or the others would be counted too